### PR TITLE
compare to null with ==

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

**Comment by a14n** [comment]
>Does dart-lang/sdk#33410 need to be resolved as well?

I think dart-lang/sdk#33410 can be closed.

**Comment by jonahwilliams** [comment]
SG, let's also make sure this doesn't break in g3.

File: packages/flutter/lib/src/painting/box_decoration.dart [hunk 1]
```diff
@@ -83,9 +83,8 @@ class BoxDecoration extends Decoration {
     this.backgroundBlendMode,
     this.shape = BoxShape.rectangle,
   }) : assert(shape != null),
-       // TODO(mattcarroll): Use "backgroundBlendMode == null" when https://github.com/dart-lang/sdk/issues/34180 is in.
        assert(
-         identical(backgroundBlendMode, null) || color != null || gradient != null,
+         backgroundBlendMode == null || color != null || gradient != null,
          'backgroundBlendMode applies to BoxDecoration\'s background color or '
          'gradient, but no color or gradient were provided.'
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/box_decoration.dart]: while you're here, can you change were to was here? thanks

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/painting/box_decoration.dart (hunk 1 above)
Comment (a14n) [reply packages/flutter/lib/src/painting/box_decoration.dart]: Sure, done!

**Pull request merged**